### DISCUSSION TITLE
Improve abs() function

### DISCUSSION THREAD

**Discussion by KemGus** [wiremod/wire#2802] [Suggestions]
Why have a separate function (positive) instead of improving abs?

The issue:
you can't return positive vector, without doing either
1. Vector=positive(vector rv1)
2. Vector = (abs(vector:x(), abs(vector:y(), abs(vector:z())

I want it to be
Vector = abs(vector)

**Comment by thegrb93** [comment]
The absolute value of a vector mathematically is it's length, so I guess they didn't want to trip up people who would expect abs(vec) to return the length.

**Comment by vurvdev** [comment]
This would be adding an overload `abs(vector v)`, not improving the existing `abs(number n)`.

Doing this would warrant deprecating `positive` and the potential name confusion from Sparky's message above. Gates also have "positive" instead of "abs(v)" so I think it's fine keeping it as is.

**Comment by vurvdev** [comment]
Also this would be a suggestion, so..
